src/optimizer/deliminator.cpp

```diff
@@ -294,4 +365,55 @@ bool Deliminator::RemoveInequalityJoinWithDelimGet(LogicalComparisonJoin &delim_
 	return found_all;
 }
 
+void Deliminator::TrySwitchSingleToLeft(LogicalComparisonJoin &delim_join) {
+	D_ASSERT(delim_join.join_type == JoinType::SINGLE);
+
+	// Collect RHS bindings
+	vector<ColumnBinding> join_bindings;
+	for (const auto &cond : delim_join.conditions) {
+		if (!IsEqualityJoinCondition(cond)) {
+			return;
+		}
+		if (cond.right->type != ExpressionType::BOUND_COLUMN_REF) {
+			return;
+		}
+		auto &colref = cond.right->Cast<BoundColumnRefExpression>();
+		join_bindings.emplace_back(colref.binding);
+	}
+
+	// Now try to find an aggr in the RHS such that the join_column_bindings is a superset of the groups
+	reference<LogicalOperator> current_op = *delim_join.children[1];
+	while (current_op.get().type != LogicalOperatorType::LOGICAL_AGGREGATE_AND_GROUP_BY) {
```

Comment:
this probably would never come up, but is it possible that there are multiple aggregates? If so, it's possible that we just check the first aggregate, but not the second.

Reply:
Well-spotted, only one of the (possibly chained) aggregates would need to have a subset of the join columns as groups